src/main/java/seedu/taassist/storage/JsonSerializableTaAssist.java

```diff
@@ -68,11 +72,21 @@ public TaAssist toModelType() throws IllegalValueException {
             if (taAssist.hasStudent(student)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_STUDENT);
             }
-            for (ModuleClass moduleClass : student.getModuleClasses()) {
+
+            // Check if the module data are valid
+            for (StudentModuleData moduleData : student.getModuleData()) {
+                ModuleClass moduleClass = moduleData.getModuleClass();
                 if (!taAssist.hasModuleClass(moduleClass)) {
                     throw new IllegalValueException(MESSAGE_CLASS_NOT_FOUND);
                 }
+                ModuleClass realModuleClass = taAssist.findModuleClass(moduleClass);
+                for (SessionData sessionData : moduleData.getSessionData()) {
+                    if (!realModuleClass.hasSession(sessionData.getSession())) {
+                        throw new IllegalValueException(MESSAGE_SESSION_NOT_FOUND);
+                    }
+                }
             }
```

Comment:
Method seem quite nested, might want to refactor the checking as a public method that is more reusable as well. Possible to do this in another issue.